frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx

```diff
@@ -178,35 +171,11 @@ const LogsTabForPodName: React.FC<{ podName: string }> = ({ podName }) => {
               </>
             ) : null}
             {podContainers.length !== 0 ? (
-              <Dropdown
-                toggle={
-                  isSmallScreen() ? (
-                    <KebabToggle
-                      onToggle={() => setIsDownloadDropdownOpen(!isDownloadDropdownOpen)}
-                    />
-                  ) : (
-                    <DropdownToggle
-                      id="download-steps-logs-toggle"
-                      onToggle={() => setIsDownloadDropdownOpen(!isDownloadDropdownOpen)}
-                    >
-                      Download
-                    </DropdownToggle>
-                  )
-                }
-                isOpen={isDownloadDropdownOpen}
-                isPlain={isSmallScreen()}
-                dropdownItems={[
-                  <DropdownItem key="current-container-logs" onClick={onDownload}>
-                    <Truncate
-                      content={isSmallScreen() ? 'Download current step log' : 'Current step log'}
-                    />
-                  </DropdownItem>,
-                  <DropdownItem key="all-container-logs" onClick={onDownloadAll}>
-                    <Truncate
-                      content={isSmallScreen() ? 'Download all step logs' : 'All step logs'}
-                    />
-                  </DropdownItem>,
-                ]}
+              <DownloadDropdown
+                onDownload={onDownload}
+                onDownloadAll={onDownloadAll}
+                isSmallScreen={isSmallScreen()}
+                isSingleStepLogsEmpty={logs === null || logs === ''}
```

Comment:
@yih-wang I remember you bringing up an empty file download if there are no logs in [this comment](https://github.com/opendatahub-io/odh-dashboard/pull/1724#issuecomment-1700660641), currently, I'm disabling the download button(single step) if there is are no logs to prevent downloading empty files.
![Screenshot from 2023-11-24 20-36-16](https://github.com/opendatahub-io/odh-dashboard/assets/22885912/abae1f1c-130e-499a-851c-5c6d47f61511)
In the case of multi-steps, the Download Dropdown is enabled but only the dropdown item corresponding to single-step logs(which is empty) is disabled.

Reply:
Another way to handle this maybe to just allow the download and giving a warning that they might be downloading an empty file.

Reply:
@manaswinidas What you have now (disabling the download action for the step that doesn't have logs) looks good.